classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace PKP\core;
+
+use PKP\core\PKPContainer;
+use Illuminate\Http\Request;
+use PKP\middlewares\HasUser;
+use PKP\middlewares\HasRoles;
+use Illuminate\Routing\Router;
+use PKP\middlewares\HasContext;
+use Illuminate\Routing\UrlGenerator;
+use PKP\middlewares\AllowCrossOrigin;
+use PKP\middlewares\ValidateCsrfToken;
+use PKP\middlewares\RemoveTrailingSlash;
+use Psr\Http\Message\ServerRequestInterface;
+use Illuminate\Routing\RoutingServiceProvider;
+use PKP\middlewares\DecodeApiTokenWithValidation;
+use PKP\middlewares\SetupContextBasedOnRequestUrl;
+use Illuminate\Foundation\Http\Middleware\TrimStrings;
+use Illuminate\Foundation\Http\Middleware\ValidatePostSize;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull;
+use Illuminate\Contracts\Routing\UrlGenerator as UrlGeneratorContract;
+
+class PKPRoutingProvider extends RoutingServiceProvider
+{
+    protected static $globalMiddlewares = [
+        AllowCrossOrigin::class,
+        SetupContextBasedOnRequestUrl::class,
+        DecodeApiTokenWithValidation::class,
+        ValidateCsrfToken::class,
+        RemoveTrailingSlash::class,
+        ValidatePostSize::class,
+        TrimStrings::class,
+        ConvertEmptyStringsToNull::class,
+    ];
+
+    /**
+     * The application's route middleware.
+     * These middleware can/should be assigned to specific routes or routes groups individually.
+     */
+    protected array $routeMiddlewares = [
+        'has.roles'     => HasRoles::class,
+        'has.user'      => HasUser::class,
+        'has.context'   => HasContext::class,
+    ];
+
+    public static function getGlobalRouteMiddlewares(): array
+    {
+        return self::$globalMiddlewares;
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->registerRouter();
+        $this->registerRouteMiddlewares();
+        $this->registerRoutePatterns();
+        $this->registerUrlGenerator();
+        $this->registerRedirector();
+        $this->registerResponseBindings();
+        $this->registerPsrRequest();
+        $this->registerPsrResponse();
+        $this->registerResponseFactory();
+        $this->registerCallableDispatcher();
+        $this->registerControllerDispatcher();
+    }
+
+    public function registerRouter(): void
+    {
+        $this->app->singleton('router', function ($app) {
+            $request = Request::capture();
+            $app->instance(\Illuminate\Http\Request::class, $request);
+            return new Router($app['events'], $app);
+        });
+    }
+
+    public function registerRouteMiddlewares(): void
```

Comment:
Middleware is plural already -- please change anywhere that refers to `Middlewares` to `Middleware` instead (unless that's a Laravel convention).

Reply:
> Middleware is plural already

That is correct . Update